src/createElementFunction.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import Relay from 'react-relay';
+
+import * as Utils from './containerUtils';
+import QueryAggregator from './QueryAggregator';
+import getRouteParams from './getRouteParams';
+import RouteGenerator from './RouteGenerator';
+
+export function createElementFunction() {
+  const queryAggregator = new QueryAggregator();
+  const routeGenerator = new RouteGenerator();
+
+  return function createElement(Component, props) {
+    const {branch, route, params, location, routeParams} = props;
+
+    // Wrap any Relay.Containers
+    let element = !Relay.isContainer(Component) ? <Component {...props} /> :
+      Utils.createContainerElement(
+        Component,
+        props,
+        getRouteParams(routeParams, location.query, route.queryParams || []),
+        route.rootContainerProps || {},
+        routeGenerator.getRouteFor(branch),
+        queryAggregator
+      );
+
+    // Wrap the root component in a RootContainer
+    element = route !== branch[0] ? element :
+      Utils.createBatchedRelayContainer(
+        element,
+        getRouteParams(params, location.query, route.queryParams || []),
```

Comment:
The problem is that the root-most container needs to have all the parameters to be able to issue its queries, since the batched queries will try to use those parameters.

If one of my nested routes has e.g. `?color=blue`, I need to pass that in as the `color` parameter to the root-most query, which isn't happening right now.